### PR TITLE
Migrate build logic to Kotlin Multiplatform

### PR DESCRIPTION
**Pull request by aSemy**
Updates the build config to use Kotlin Multiplatform, however the only target is Kotlin JVM

Initial step for #1. 

### TODO

- [x] ~Fix signing config~ to be completed in #4

### CODE REVIEW

**Comment by xn32** [comment]
Thank you for the draft!

If you want, feel free to remove everything related to signing and publication on Maven Central for the time being. We can add it again when everything else is ready and thoroughly tested.

**Comment by aSemy** [comment]
@xn32 It's ready for review then :) I'm working on the publishing config now, I can put it in another PR

File: build.gradle.kts [hunk 1]
```diff
@@ -1,66 +1,77 @@
...
     kotlin("plugin.serialization") version "1.7.20"
     id("org.jetbrains.kotlinx.kover") version "0.6.1"
     id("org.jetbrains.dokka") version "1.7.20"
     `maven-publish`
-    `java-library`
     signing
 }
 
 group = "io.github.xn32"
 version = "0.2.1"
```
Comment (xn32) [inline build.gradle.kts]: Let's start to prepare for version 0.3.0:
```suggestion
version = "0.3.0-SNAPSHOT"
```

File: build.gradle.kts [hunk 2]
```diff
@@ -1,66 +1,77 @@
...
```
Comment (xn32) [inline build.gradle.kts:18]: For the sake of simplicity, I would like to keep the build logic in this file for now:
```suggestion
repositories {
    mavenCentral()
}

```

File: build.gradle.kts [hunk 3]
```diff
@@ -82,25 +93,17 @@ publishing {
...
-
 signing {
     isRequired = isReleaseVersion && gradle.taskGraph.hasTask("publish")
-    sign(publishing.publications["mavenJava"])
-}
-
-tasks.withType<KotlinCompile> {
-    kotlinOptions.jvmTarget = "1.8"
+    // TODO updating signing
+//    sign(publishing.publications["mavenJava"])
```
Comment (xn32) [inline build.gradle.kts]: This should do the trick, I think:
```suggestion
    sign(publishing.publications)
```

File: settings.gradle.kts [hunk 4]
```diff
@@ -1 +1,18 @@
...
+    mavenCentral()
+  }
+
+  pluginManagement {
+    repositories {
+      gradlePluginPortal()
+      mavenCentral()
+    }
+  }
+}
```
Comment (xn32) [inline settings.gradle.kts]:
```suggestion
```

**Review by xn32 — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 3 above)
Comment (aSemy) [reply build.gradle.kts]: I'm not so confident, because the signing plugin is very buggy :) But it will be fixed in #4

**Pull request merged**